brian2tools/baseexport/device.py

```diff
@@ -0,0 +1,417 @@
+                                           'in device mode, '
+                                           'try using string '
+                                           'expressions') % variableview.name)
+        self.initializers_connectors.append(init_dict)
+
+    def synaptic_pathway_before_run(self, pathway, run_namespace):
+        """
+        Get synaptic pathways and pass it to dictionary
+        """
+        # handled in the collect_synapses()
+        pass
+
+    def synapses_connect(self, synapses, condition=None, i=None,
+                         j=None, p=1, n=1, skip_if_invalid=False,
+                         namespace=None, level=0):
+        """
+        Override synapses_connect() to get details from Synapses.connect()
+        """
+        # if namespace not defined
+        if namespace is None:
+            namespace = get_local_namespace(level + 2)
+        # prepare objects using `before_run()`
+        synapses.before_run(namespace)
+        if condition is not None:
+            if i is not None or j is not None:
+                raise ValueError("Cannot combine condition with i or j "
+                                 "arguments")
+        connect = {}
+        # string statements that shall have identifers
+        strings_with_identifers = []
+        if condition:
+            connect.update({'condition': condition})
+            strings_with_identifers.append(condition)
+        elif i is not None or j is not None:
+            if i is not None:
+                connect.update({'i': i})
+            if j is not None:
+                connect.update({'j': j})
+        connect.update({'probability': p, 'n_connections': n,
+                        'synapses': synapses.name,
+                        'source': synapses.source.name,
+                        'target': synapses.target.name, 'type': 'connect'
+                        })
+        # get resolved and clean identifiers
+        strings_with_identifers.append(str(p))
+        strings_with_identifers.append(str(n))
+        identifers_set = set()
+        for string_expr in strings_with_identifers:
+            identifers_set = identifers_set | get_identifiers(string_expr)
+        ident_dict = synapses.resolve_all(identifers_set, namespace)
+        ident_dict = _prepare_identifiers(ident_dict)
+        if ident_dict:
+            connect.update({'identifiers': ident_dict})
+        self.initializers_connectors.append(connect)
+        # update `_connect_called` to allow initialization on
+        # synaptic variables
+        synapses._connect_called = True
+
+    def build(self, direct_call=True, debug=False):
+        """
+        Collect all run information and export the standard
+        dictionary format
+
+        Parameters
+        ----------
+        direct_call: bool, optional
+            To check the call to build() was made directly
+
+        debug: bool, optional
+            To build the device in debug mode
```

Comment:
I'd add a half-sentence here saying what this actually means ("... which prints out the dictionary information")